frontend/src/ui/pages/RegisterPage/RegisterPage.js

```diff
@@ -0,0 +1,139 @@
+		username: Yup.string().required('Wait! What\'s your username?')
+	})
+
+	const handleSubmit = async (values) => {
+		const avatar = {
+			'body': /^([^.]+)/.exec(Object.keys(body)[currentBody])[1],
+			'hair': /^([^.]+)/.exec(Object.keys(hair)[currentHair])[1],
+			'fit': /^([^.]+)/.exec(Object.keys(fit)[currentFit])[1]
+		}
+
+		const user = {...values, hairID: parseInt(avatar.hair),  outfitTopID: parseInt(avatar.fit), outfitBottomID: parseInt(avatar.body)}
+
+		const res = await register(user)
+		if (!res) {
+			setErrors(['Server not available'])
+		} else {
+			if (res.status === 201) {
+				setToggle(true)
+				setTimeout(() => {
+					navigate('/login')
+				}, 3000)
+			} else if (res.status === 400) {
+				setTimeout(() => {
+					setErrors([])
+				}, 4000)
+				setErrors([res.data.message, ...errors])
+			}
+		}
+	}
+
+	function importAll(r) {
+		let images = {}
+		r.keys().forEach((item) => { images[item.replace('./', '')] = r(item) })
+		return images
+	}
+    
+	// eslint-disable-next-line no-undef
+	const body = importAll(require.context('../../assets/avatar_files/body', false, /\.(png|jpe?g|svg)$/))
+	// eslint-disable-next-line no-undef
+	const fit = importAll(require.context('../../assets/avatar_files/fit', false, /\.(png|jpe?g|svg)$/))
+	// eslint-disable-next-line no-undef
+	const hair = importAll(require.context('../../assets/avatar_files/hair', false, /\.(png|jpe?g|svg)$/))
+
+	useEffect(() => {
+
+	}, [currentBody, currentFit, currentHair])
+
+
+	return (
+		<div className={`register-page ${toggle ? 'overlay' : null}`}> 
+			{(toggle) ? 
+				<RegisterPopup /> : null
+			}
+			<div className='register-container'>
+				<Formik
+					initialValues={{ username: '', password: '' , passwordConfirmation: ''}}
+					validationSchema={registerSchema}
+					onSubmit={(values, { resetForm }) => {return (handleSubmit(values), resetForm())}}>
+					{() => {
+						return (
+							<Form className='register-form'>
+								<p className='header'>SIGNUP</p>
+								<Text className='register-field-box'>
+									<Field className="register-field" placeholder="USERNAME" type="username" name="username" />
+									<ErrorMessage className='error-msg' name="username" component="div" />
+								</Text>
+								<Text className='register-field-box'>
+									<Field className="register-field" placeholder="PASSWORD" type="password" name="password" />
+									<ErrorMessage className='error-msg' name="password" component="div" />
+								</Text>
+								<Text className='register-field-box'>
+									<Field className="register-field" placeholder="CONFIRM YOUR PASSWORD" type="password" name="passwordConfirmation" />
+									<ErrorMessage className='error-msg' name="passwordConfirmation" component="div" />
+									{errors ? errors.map(e => <div key={e} className='error-msg'>{e}</div>) : null}
+								</Text>
+								<Button className='register-btn'><button className='register-btn btn' type="submit">CREATE MY ACCOUNT</button></Button>
```

Comment:
why you placed a button inside a button here? I think you can just do something like this:
`<Button className='register-btn'>CREATE MY ACCOUNT</Button>`

Reply:
because <Button /> is simply for style purposes and we cannot perform submit operatio on it

Reply:
Are you sure? Can't we simply add an additional prop 'type' to the Button component? @kbojarowska, what do you think?

Reply:
From my perspective the Button compoent was created to keep the same theme in styling - more like a container. I don't think it's a major inconvenience to use an additional button inside it.

Reply:
I think Julia is right, it would be better to add non obligatory prop 'type' which could be added or not, according to needs @zdwo @juliadlutek

Reply:
done